bindgen-integration/build.rs

```diff
@@ -59,15 +64,64 @@ impl ParseCallbacks for MacroCallback {
             _ => {}
         }
     }
+
+    fn func_macro(&self, name: &str, value: &[&[u8]]) {
+        match name {
+            "TESTMACRO_NONFUNCTIONAL" => {
+                panic!("func_macro was called for a non-functional macro");
+            }
+            "TESTMACRO_FUNCTIONAL_NONEMPTY(TESTMACRO_INTEGER)" => {
+                // Spaces are inserted into the right-hand side of a functional
+                // macro during reconstruction from the tokenization. This might
+                // change in the future, but it is safe by the definition of a
+                // token in C, whereas leaving the spaces out could change
+                // tokenization.
+                assert_eq!(
+                    value,
+                    &["-".as_bytes(), "TESTMACRO_INTEGER".as_bytes()]
+                );
+                *self.seen_funcs.lock().unwrap() += 1;
+            }
+            "TESTMACRO_FUNCTIONAL_EMPTY(TESTMACRO_INTEGER)" => {
+                assert_eq!(value, &[] as &[&[u8]]);
+                *self.seen_funcs.lock().unwrap() += 1;
+            }
+            "TESTMACRO_FUNCTIONAL_TOKENIZED(a,b,c,d,e)" => {
+                assert_eq!(
+                    value,
+                    ["a", "/", "b", "c", "d", "##", "e"]
```

Comment:
My verbose formulation was the result of my giving up after failing to bend the type inference to my will. It would work fine if we had all one-character strings, but as soon as the `b"string"`s vary in length, [type inference for an array of them fails](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=49384eaf2fa499063b9fec0968f04a35). I assume that `&str` is special-cased in a way that `&[u8]` is not. Maybe I am missing something simple, though.

Reply:
I figured out what I needed to do :
```rust
&[ b"a" as &[u8], b"/", ... ]
```